app/scripts/views/permissions.js

```diff
@@ -30,20 +70,163 @@ define(function (require, exports, module) {
 
     context: function () {
       var account = this.getAccount();
+      var requestedPermissions = this.relier.get('permissions');
+      var applicablePermissions =
+        this._getApplicablePermissions(account, requestedPermissions);
+      var permissionsHTML = this._getPermissionsHTML(account, applicablePermissions);
+
       return {
-        email: account.get('email'),
         privacyUri: this.relier.get('privacyUri'),
         serviceName: this.relier.get('serviceName'),
-        termsUri: this.relier.get('termsUri')
+        termsUri: this.relier.get('termsUri'),
+        unsafePermissionsHTML: permissionsHTML,
       };
     },
 
+    /**
+     * Get configuration for a permission
+     *
+     * @private
+     * @param {string} permissionName
+     * @returns {object} permission, if found.
+     * @throws if permission is invalid
+     */
+    _getPermissionConfig: function (permissionName) {
+      var permission = _.findWhere(PERMISSIONS, { name: permissionName });
+
+      if (! permission) {
+        // log the invalid scope instead of throwing an error
+        // to see if any reliers are specifying invalid scopes. We
+        // will be more strict in the future. Ref #2508
+        this.logViewEvent('scope.invalid', permissionName);
```

Comment:
It seems like the `logViewEvent` should be moved to `_getApplicablePermissions` to avoid making assumptions about where `_getPermissionConfig` is called from. Otherwise, `_getPermissionConfig` could be called from multiple places with the same invalid permission and it'll be logged multiple times.